src/lib/rest/rest.cpp

```diff
@@ -616,6 +630,17 @@ static int contentTypeCheck(ConnectionInfo* ciP)
     return 1;
   }
 
+
+  // Case 5
+  if ((ciP->apiVersion == "v2") && (ciP->httpHeaders.contentType != "application/json"))
```

Comment:
Being stricts, this is not part of #947 (as the operation described in that issue number is a GET without payload). I understand that you are preparing the code in advance for the operations that are to come and that use payload. Is that correct?